File: android/app/src/main/java/com/okhttpandroidapp/CustomNetworkModule.java
package com.okhttpandroidapp;

import com.facebook.react.modules.network.OkHttpClientFactory;

import okhttp3.OkHttpClient;

class CustomNetworkModule implements OkHttpClientFactory {
    public OkHttpClient createNewNetworkModuleClient() {
        return new OkHttpClient.Builder()
                .build();
    }
}

File: android/app/src/main/java/com/okhttpandroidapp/okhttp/CustomNetworkModule.kt
package com.okhttpandroidapp.okhttp

import com.facebook.react.modules.network.OkHttpClientFactory
import com.facebook.react.modules.network.ReactCookieJarContainer
import com.okhttpandroidapp.networks.NetworksLiveData
import okhttp3.ConnectionPool

import okhttp3.OkHttpClient

internal class CustomNetworkModule : OkHttpClientFactory {
    val connectionPool = ConnectionPool()
    val cookieJarContainer = ReactCookieJarContainer()
    val client = OkHttpClient.Builder()
            .connectionPool(connectionPool)
            .cookieJar(cookieJarContainer)
            .build()

    override fun createNewNetworkModuleClient(): OkHttpClient {
        return client
    }
}